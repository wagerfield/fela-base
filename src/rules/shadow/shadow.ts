import { Unit } from "../../types"
import { getRuleProps } from "../defaults"

export interface ShadowRuleProps {
  x?: Unit
  y?: Unit
  blur?: Unit
  spread?: Unit
  color?: string
}

export const NAME = "shadow"
export const DEFAULTS = {
  x: 0,
  y: "4px",
  blur: "8px",
  spread: 0,
  color: "rgba(0,0,0,0.5)"
}

export default (props?: ShadowRuleProps) => {
  const p = getRuleProps(NAME, DEFAULTS, props)
  return {
    boxShadow: `${p.x} ${p.y} ${p.blur} ${p.spread} ${p.color}`
  }
}
