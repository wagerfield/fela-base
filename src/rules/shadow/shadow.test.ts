import shadow from "./shadow"

describe("shadow(props)", () => {
  it("renders expected snapshots", () => {
    expect(shadow()).toMatchSnapshot()
    expect(
      shadow({
        x: "4px",
        spread: "8px"
      })
    ).toMatchSnapshot()
  })
})
