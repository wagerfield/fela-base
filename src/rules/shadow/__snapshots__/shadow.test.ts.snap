// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shadow(props) renders expected snapshots 1`] = `
Object {
  "boxShadow": "0 4px 8px 0 rgba(0,0,0,0.5)",
}
`;

exports[`shadow(props) renders expected snapshots 2`] = `
Object {
  "boxShadow": "4px 4px 8px 8px rgba(0,0,0,0.5)",
}
`;
